tools/shell-commands/src/main/java/org/apache/unomi/shell/migration/impl/MigrationTo200.java

```diff
@@ -126,25 +113,88 @@ private void copyValueScopeToSourceId(final String indexName, final CloseableHtt
         httpPost.addHeader("Accept", "application/json");
         httpPost.addHeader("Content-Type", "application/json");
 
-        String request = "{\n" +
-                "  \"script\": {\n" +
-                "    \"source\": \"ctx._source.sourceId = ctx._source.scope\",\n" +
-                "    \"lang\": \"painless\"\n" +
-                "  }\n" +
-                "}";
+        String request = "{\n" + "  \"script\": {\n" + "    \"source\": \"ctx._source.sourceId = ctx._source.scope\",\n"
+                + "    \"lang\": \"painless\"\n" + "  }\n" + "}";
 
         httpPost.setEntity(new StringEntity(request));
 
         try (CloseableHttpResponse response = httpClient.execute(httpPost)) {
             JSONObject responseAsJson = new JSONObject(EntityUtils.toString(response.getEntity()));
 
             if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
-                System.out.println("Copying the \"scope\" field to the \"sourceId\" field for index = \"" + indexName + "\" successfully completed. Total: " +
-                        responseAsJson.get("total") + ", updated: " + responseAsJson.get("updated") + ".");
+                System.out.println("Copying the \"scope\" field to the \"sourceId\" field for index = \"" + indexName
+                        + "\" successfully completed. Total: " + responseAsJson.get("total") + ", updated: " + responseAsJson.get("updated")
+                        + ".");
             } else {
                 System.out.println("Copying the \"scope\" field to the \"sourceId\" field for index = \"" + indexName + "\" failed.");
             }
         }
     }
 
+    private Set<String> getEventIndexes(String indexPrefix) throws IOException {
+        try (CloseableHttpResponse response = httpClient.execute(new HttpGet(esAddress + "/_aliases"))) {
+            if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+                JSONObject indexesAsJson = new JSONObject(EntityUtils.toString(response.getEntity()));
+                return indexesAsJson.keySet().stream().
+                        filter(alias -> alias.startsWith(indexPrefix + "-event")).
+                        collect(Collectors.toSet());
+            }
+        }
+        return Collections.emptySet();
+    }
+
+    private void createScopes(Set<String> scopes, String indexPrefix) throws IOException {
+        final StringBuilder body = new StringBuilder();
```

Comment:
Careful here it seem's you forgot to create the scope index with the mapping provided in:
persistence-elasticsearch/core/src/main/resources/META-INF/cxs/mappings/scope.json